### PR TITLE
fix: endpoints in dev server should never have trailing slash

### PR DESCRIPTION
**Pull request by ematipico**
## Changes

Closes https://github.com/withastro/astro/issues/10149

## Testing

Added a new test case. It seems we didn't have one.

### CODE REVIEW

File: .changeset/dry-suns-protect.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"astro": patch
+---
+
+Fixes a regression where the dev server was returning a 404 when hitting an endpoint, when the app was configured with `trailingSlash: "never"`
```
Comment (lilnasy) [inline .changeset/dry-suns-protect.md:5]: Did you mean "always"?

Comment (ematipico) [reply .changeset/dry-suns-protect.md:5]: I did!

**Review by matthewp — APPROVED** [review]

**Review by lilnasy — CHANGES_REQUESTED** [review]
Assuming the changeset needs to be corrected, I think this is the wrong call.

Users should configure "trailingSlash" to inform the dev server that the production server will behave a certain way so it should too. If the behavior doesn't match, they may need to configure something else. `trailingSlash` is doing what the config is telling it to.

**Comment by matthewp** [comment]
Hm, that's a good point, let's chat about it.

**Comment by lilnasy** [comment]
Test cases already exist in [test/units/routing/trailing-slash.test.js](https://github.com/withastro/astro/pull/9597/files#diff-859cc2635b0ceffafe57a59e8c20d9c4fb5b5cb70b9245c430e5335b1b2b46ee)

**Comment by kripod** [comment]
I‘m afraid this issue still persists: https://github.com/withastro/astro/issues/10149#issuecomment-1961767578

[Static File Endpoints](https://docs.astro.build/en/guides/endpoints/#static-file-endpoints) exposing e.g. `.png` assets for dynamic social image generation can‘t be accessed as `/img.png` in dev mode, as only `/img.png/` would pass the overly strict trailing slash requirements.

**Pull request closed without merging**